### PR TITLE
Add function for learning_phase default config

### PR DESCRIPTION
**Pull request by Nycroth**
This is a follow-on to a previous pull request that was [closed and ignored](https://github.com/keras-team/keras/pull/8913).

This issue addressed by this patch is that a production model will rarely want to export the `keras_learning_phase` placeholder to a serving interface. A train only Keras graph may be faster but it's also useless except in the rare circumstance that training features (BN, dropout, etc) are desired at evaluation and prediction time. Without a default placeholder or a fixed value the `keras_learning_phase` must be either fed as part of a prediction request (exposing it and adding unneeded complexity to the interface), reprogram the serving endpoint to transparently feed this (essentially a hack, and possibly not possible), or have the pleasure of performing TensorSurgery to convert the placeholder to a constant on the fly into the SavedModel protobuffer (a laborious and unpleasant hack). Beyond this, the only other option is to rewrite without Keras, something we've had to do with our current model due to this current limitation.

### CODE REVIEW

**Comment by fchollet** [comment]
The idea is that you can train your models with a training-only graph, and put them in production with a testing-only graph (one that doesn't include at all dropout ops, for instance). By simply setting `K.set_learning_phase(1)` (or `0`) before building the graph.

This is in fact the standard practice with most TensorFlow workflows.
